cdc_replication_test.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2020 ScyllaDB
+
+import shutil
+import sys
+import os
+import time
+from tempfile import TemporaryDirectory
+from textwrap import dedent
+
+from cassandra import ConsistencyLevel
+from cassandra.query import SimpleStatement
+
+from sdcm.tester import ClusterTester
+
+def print_file_to_stdout(path: str) -> None:
+    with open(path, "r") as f:
+       shutil.copyfileobj(f, sys.stdout)
+
+# Write a CQL select result to a file.
+def write_cql_result(res, path: str):
+    with open(path, 'w') as f:
+        for r in res:
+            f.write(str(r) + '\n')
+        f.flush()
+        os.fsync(f.fileno())
+
+SCYLLA_MIGRATE_URL = "https://kbr-scylla.s3-eu-west-1.amazonaws.com/scylla-migrate"
+REPLICATOR_URL = "https://kbr-scylla.s3-eu-west-1.amazonaws.com/scylla-cdc-replicator-0.0.1-SNAPSHOT-jar-with-dependencies.jar"
+
+def keyspace_def(rf: int) -> str:
+    return f"create keyspace if not exists ks with replication = {{'class': 'SimpleStrategy', 'replication_factor': {rf}}}"
+
+TABLE_DEF_BASE = "create table if not exists ks.tb (pk int, ck int, v text, primary key (pk, ck))"
+TABLE_DEF_MASTER = TABLE_DEF_BASE + " with cdc = {'enabled': true}"
+
+class CDCReplicationTest(ClusterTester):
+    def test_replication(self):
+        print("Adding node to db_cluster... ")
+        node2 = self.db_cluster.add_nodes(count=1, enable_auto_bootstrap=True)[0]
+        print("Waiting for added node to initialize...")
+        self.db_cluster.wait_for_init(node_list=[node2])
+
+        print("Adding node to db_cluster... ")
+        node3 = self.db_cluster.add_nodes(count=1, enable_auto_bootstrap=True)[0]
+        print("Waiting for added node to initialize...")
+        self.db_cluster.wait_for_init(node_list=[node3])
+
+        master_node = self.db_cluster.nodes[0]
+        replica_node = self.cs_db_cluster.nodes[0]
+
+        print('Creating table on master cluster.')
+        with self.cql_connection_patient(node=master_node) as sess:
+            sess.execute(keyspace_def(3))
+            sess.execute(TABLE_DEF_MASTER)
```

Comment:
i think it is not need on master cluster. Because cassandra will create it automatically from profile.

Reply:
I actually removed the keyspace definition from the profile to reduce duplication. I wanted to have the KS definition in a single place. Since I need to have it in the code anyway (to create the KS on the second cluster), I decided to put it here.